amida-tech/greyscale
# Feature/inba 969 jf all groups endpoints

@au2 I could use some help on this one. Due to the inclusion of an array of user ids being returned on the `/:realm/v0.2/organizations/2/groups` endpoint, the tests are failing and due to the heavy automation, I'm at a loss as to how to adjust. 

The failing tests is in `backend\test\group.integration.js`, lines 99, 105, 109, referencing `tests.listGroupsFn()` under `group-common.js`. If you have any insight, that would be helpful. 

#### What does this PR do?
Modifies the groups endpoint to return project information when listing all groups. 

#### Related JIRA tickets:
INBA-969
INBA-959

#### Did the user.integration, discussion.integration and task.integration tests pass? Test can be run with:
#### node_modules/.bin/mocha test/user.integration.js test/discussion.integration.js test/task.integration.js --bail -t 10000

#### How should this be manually tested?
In your `.env`, be sure to set `GREYSCALE_PG_TEST_DB` to the name of whatever database you want to run your testing from. I recommend `indaba_test`.
Be sure you have a database named as such (`indaba_test`)
Run `node_modules/.bin/mocha test/group.integration.js --bail -t 10000`

To run manually, get the Bearer token for an admin and run a POST call against `http://localhost:3005/develop/v0.2/organizations/2/groups` then confirm you get an array of objects like:
`{
        "projectId": x,
        "projectName": "Whatever",
        "id": x,
        "title": "Whatever the group title is",
        "organizationId": x,
        "langId": null,
        "userIds": [
            x,
            y
        ]
},`

#### Background/Context
Per suggestions on how we're going to handle group searches, group listings return the projects they're currently associated with when the user messes with User Groups. 

#### Screenshots (if appropriate):
